test/letters/actions/letters.unit.spec.js

```diff
@@ -359,49 +370,61 @@ describe('getBenefitSummaryOptions', () => {
-  //     serviceConnectedEvaluation: true,
-  //     chapter35Eligibility: true,
-  //     serviceConnectedDisabilities: true
-  //   }
-  // };
-
-  // const benefitSLetter = {
-  //   letterName: 'Benefit Summary Letter',
-  //   letterType: LETTER_TYPES.benefitSummary,
-  //   letterOptions: {
-  //     // Opts only relevant for BSL but ATM required in every download link
-  //     militaryService: true,
-  //     monthlyAward: true,
-  //     serviceConnectedEvaluation: true,
-  //     chapter35Eligibility: true,
-  //     serviceConnectedDisabilities: true
-  //   }
-  // };
-
-  it('sets up fetch for benefit summary letter', () => {});
-
-  it('sets up fetch for non-benefit-summary letters', () => {});
-
-  it('dispatches PENDING action when download initiated', () => {});
-
-  it('handles ie10 stuff', () => {});
-
-  it('downloads stuff conditionally', () => {});
-
-  it('dispatches DOWNLOAD_SUCCESS once download succeeds', () => {});
-
-  it('dispatches DOWNLOAD_FAILED if download or fetch fails', () => {});
+  const benefitSLetter = {
+    letterName: 'Benefit Summary Letter',
+    letterType: LETTER_TYPES.benefitSummary,
+    letterOptions: {
+      militaryService: true,
+      monthlyAward: true,
+      serviceConnectedEvaluation: true,
+      chapter35Eligibility: true,
+      serviceConnectedDisabilities: true
+    }
+  };
+
+  it('dispatches download pending action first', (done) => {
+    const { letterType, letterName, letterOptions } = benefitSLetter;
+    const thunk = getLetterPdf(letterType, letterName, letterOptions);
+    const dispatch = sinon.spy();
+    thunk(dispatch, getState)
+      .then(() => {
+        const action = dispatch.firstCall.args[0];
+        expect(action.type).to.equal(GET_LETTER_PDF_DOWNLOADING);
+        expect(action.data).to.equal(letterType);
+      }).then(done, done);
+  });
+
+  it('dispatches SUCCESS action when fetch succeeds', (done) => {
+    const { letterType, letterName, letterOptions } = benefitSLetter;
+    global.fetch.returns(Promise.resolve({
+      headers: { get: () => 'application/octet-stream' },
+      ok: true,
+      blob: () => Promise.resolve({ test: '123 testing' })
+    }));
+    const thunk = getLetterPdf(letterType, letterName, letterOptions);
+    const dispatch = sinon.spy();
+    thunk(dispatch, getState)
+      .then(() => {
+        const action = dispatch.secondCall.args[0];
+        expect(action.type).to.equal(GET_LETTER_PDF_SUCCESS);
+      }).then(done, done);
+  });
+
+  it('dispatches FAILURE action if download fails', (done) => {
+    global.fetch.returns(Promise.reject(new Error('Oops, this failed')));
+    const { letterType, letterName, letterOptions } = benefitSLetter;
+    const thunk = getLetterPdf(letterType, letterName, letterOptions);
+    const dispatch = sinon.spy();
+    thunk(dispatch, getState)
+      .then(() => {
+        const action = dispatch.secondCall.args[0];
+        expect(action.type).to.equal(GET_LETTER_PDF_FAILURE);
```

Comment:
We assume it should fail because we never configure `fetch` to return anything?

Reply:
Yeah, any sort of error like this should get caught in the `onFailure` argument of `apiRequest`, so the thinking is that instead of mocking out a response object with errors this is simpler (since we don't distinguish between error types / codes anyway).